### DISCUSSION TITLE
Feature Request: WalletConnect to handle Celo-specific identity data

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/social-connect#215] [Feature requests (Needs)]
Celo developers that want to lookup phone numbers are required to use [Celo DAppKit](https://docs.celo.org/developer-guide/dappkit#functionality). If developers want to use WalletConnect, there is no simple way to lookup phone numbers or pass identity data through WalletConnect. 

For example, [Ubeswap](https://ubeswap.org/) display a user's phone number in their UI using DAppKit. 

What's DAppKit?

> DAppKit is a lightweight set of functions that allow mobile DApps to work with the Celo Wallet to sign transactions and access the user's account. This allows for a better user experience: DApps can focus on a great native experience without having to worry about key management. It also provides a simpler development experience, as no state or connection management is necessary.
_Source: [docs.celo.org](https://docs.celo.org/developer-guide/dappkit#functionality)_

**Comment by vibern0** [comment]
Ohh, I was not aware of this.
~This is a breaking change to us~ :eyes: :confused:

**Reply by arthurgousset** [reply]
That's good to know 💯  ~~Thanks for flagging @obernardovieira! I'll consider this as a downvote (👎).~~
It's not currently on our roadmap :)

**Reply by vibern0** [reply]
oh no, I mean, it will be great having it. We also use the phone number.
But not a breaking change actually, we've decided to change a few things.

**Reply by arthurgousset** [reply]
My bad, I misunderstood you 🙋‍♂️ Edited above.
